src/test/java/org/prebid/server/it/UnrulyTest.java

```diff
@@ -22,8 +22,12 @@
     public void openrtb2AuctionShouldRespondWithBidsFromUnruly() throws IOException, JSONException {
         // given
         WIRE_MOCK_RULE.stubFor(post(urlPathEqualTo("/unruly-exchange"))
-                .withRequestBody(equalToJson(jsonFrom("openrtb2/unruly/test-unruly-bid-request.json")))
-                .willReturn(aResponse().withBody(jsonFrom("openrtb2/unruly/test-unruly-bid-response.json"))));
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/unruly/test-unruly-bid-request-1.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/unruly/test-unruly-bid-response-1.json"))));
+
+        WIRE_MOCK_RULE.stubFor(post(urlPathEqualTo("/unruly-exchange"))
+                .withRequestBody(equalToJson(jsonFrom("openrtb2/unruly/test-unruly-bid-request-2.json")))
+                .willReturn(aResponse().withBody(jsonFrom("openrtb2/unruly/test-unruly-bid-response-2.json"))));
```

Comment:
Why do you have 2 mocks, if you have 1 imp in request -> 1 bid request?

Reply:
Yes, we have logic which `do request per imp` .
In 2 mocks separate check **Banner** and **Video** types .
But I'm thinking change `test-unruly-bid-request-2.json`, add second type.
Correct my if i am wrong.

Reply:
1 request per 1 imp logic does not necessarily mean 2 separate requests. That's the beauty of it. You can look into other bidders, some of them does that. AdPrime bidder is one of them.